azurelinuxagent/ga/update.py

```diff
@@ -387,6 +340,45 @@ def run(self, debug=False):
         self._shutdown()
         sys.exit(0)
 
+    def _check_daemon_running(self, debug):
+        # Check that the parent process (the agent's daemon) is still running
+        if not debug and self._is_orphaned:
+            logger.info("Agent {0} is an orphan -- exiting", CURRENT_AGENT)
+            raise ExitException()
+
+    def _check_threads_running(self, all_thread_handlers):
+        # Check that all the threads are still running
+        for thread_handler in all_thread_handlers:
+            if not thread_handler.is_alive():
+                logger.warn("{0} thread died, restarting".format(thread_handler.get_thread_name()))
+                thread_handler.start()
+
+    def _process_goal_state(self, protocol, exthandlers_handler, remote_access_handler):
+        if not protocol.try_update_goal_state():
+            self._heartbeat_update_goal_state_error_count += 1
+            return
+
+        if self._upgrade_available(protocol):
```

Comment:
the agent update is part of processing the goal state, will leave here

Reply:
Technically in the Linux agent we upgrade the Agent even if there's no new incarnation. I know you kept the same behavior even now but its still technically independent of the GoalState.  

If in the future we're going to change that behavior and only update the agent on new incarnations (something that Windows agent does today), then the agent update would be part of the goal state but for now it might add more confusion dont you think?

Reply:
agent update is part of processing  the goal state (except we don't care about the incarnation, and we do it every hour) and it depends on the goal state.

regardless, that is more a philosophical discussion, i guess. for the purposes of preparing for fast track i'll leave it where it currently is

thanks for the comment, though

Reply:
sure no worries, thanks for explaining :)